x/programs/rust/examples/multisig/src/lib.rs

```diff
@@ -0,0 +1,392 @@
+        .state()
+        .store_by_key(StateKeys::LastProposalId, &(last_id + 1))
+        .unwrap();
+
+    last_id
+}
+
+#[public]
+pub fn vote(context: Context<StateKeys>, id: u32, yea: bool) -> Result<(), ProposalError> {
+    let program = context.program();
+    let mut proposal = proposal_at(program, id).ok_or(ProposalError::InexistentProposal)?;
+
+    let actor = context.actor();
+    if proposal.voters.contains(&actor) {
+        return Err(ProposalError::AlreadyVoted);
+    }
+    if proposal.executed {
+        return Err(ProposalError::AlreadyExecuted);
+    }
+
+    if yea {
+        proposal.yea += 1;
+    } else {
+        proposal.nay += 1;
+    }
+    proposal.voters.insert(actor);
+
+    program
+        .state()
+        .store_by_key(StateKeys::Proposals(id), &proposal)
+        .expect("state corrupt");
+
+    Ok(())
+}
+
+#[public]
+pub fn execute(
+    context: Context<StateKeys>,
+    proposal_id: u32,
+    max_units: u64,
+) -> Result<DeferDeserialize, ProposalError> {
+    let program = context.program();
+    let mut proposal =
+        proposal_at(program, proposal_id).ok_or(ProposalError::InexistentProposal)?;
+
+    if proposal.executed {
+        return Err(ProposalError::AlreadyExecuted);
+    }
+    if !quorum_reached(&proposal) {
+        return Err(ProposalError::QuorumNotReached);
+    }
+
+    proposal.executed = true;
+
+    proposal
+        .to
+        .call_function::<DeferDeserialize>(
+            &proposal.method,
+            &proposal.data,
+            max_units,
+            proposal.value,
+        )
+        .map_err(ProposalError::ExecutionFailed)
+}
+
+#[public]
+pub fn pub_proposal_at(context: Context<StateKeys>, proposal_id: u32) -> Option<Proposal> {
+    proposal_at(context.program(), proposal_id)
+}
+
+#[public]
+pub fn pub_proposal_id(context: Context<StateKeys>) -> u32 {
+    proposal_id(context.program())
+}
+
+#[public]
+pub fn pub_quorum_reached(
+    context: Context<StateKeys>,
+    proposal_id: u32,
+) -> Result<bool, ProposalError> {
+    let proposal =
+        proposal_at(context.program(), proposal_id).ok_or(ProposalError::InexistentProposal)?;
+    Ok(quorum_reached(&proposal))
+}
+
+pub fn proposal_at(program: &Program<StateKeys>, proposal_id: u32) -> Option<Proposal> {
+    program
+        .state()
+        .get(StateKeys::Proposals(proposal_id))
+        .expect("state corrupt")
+}
+
+pub fn proposal_id(program: &Program<StateKeys>) -> u32 {
+    program
+        .state()
+        .get(StateKeys::LastProposalId)
+        .expect("state corrupt")
+        .unwrap_or_default()
+}
+
+pub fn quorum_reached(proposal: &Proposal) -> bool {
+    proposal.yea + proposal.nay >= MIN_VOTES && proposal.yea > proposal.nay
+}
+
+#[cfg(test)]
+mod tests {
+    use super::ProposalError;
+    use simulator::{Endpoint, Key, Step, TestContext};
+    use wasmlanche_sdk::DeferDeserialize;
+
+    const PROGRAM_PATH: &str = env!("PROGRAM_PATH");
+
+    #[test]
+    fn can_propose() {
```

Comment:
You're testing the IDs functionality here... "can propose" is storing state, you don't need to test whether or not you can store state.